src/compiler/emitter.ts

```diff
@@ -4649,44 +4636,78 @@ var __param = this.__param || function(index, decorator) { return function (targ
 
             function emitAMDModule(node: SourceFile, startIndex: number) {
                 collectExternalModuleInfo(node);
-                writeLine();
-                write("define(");
-                sortAMDModules(node.amdDependencies);
-                if (node.amdModuleName) {
-                    write("\"" + node.amdModuleName + "\", ");
-                }
-                write("[\"require\", \"exports\"");
+                
+                /// An AMD define function has the following shape:
+                ///     define(id?, dependencies?, factory);
+                ///
+                /// This has the shape of
+                ///     define(name, ["module1", "module2"], function (module1Alias) {
+                /// The location of the alias in the parameter list in the factory function needs to 
+                /// match the position of the module name in the dependency list.
+                ///
+                /// To ensure this is true in cases of modules with no aliases, e.g.: 
+                /// `import "module"` or `<amd-dependency path= "a.css" \>` we need to add these 
+                /// modules to the end of the dependencies list

```

Comment:
"to add these" -> "to add modules without alias names to the end"

Also note here that `aliasedModuleNames` and `importAliasNames` need to have corresponding indices for their module names and aliasing names